src/main/java/org/mockito/internal/junit/DefaultStubbingLookupListener.java

```diff
@@ -0,0 +1,57 @@
+package org.mockito.internal.junit;
+
+import org.mockito.internal.exceptions.Reporter;
+import org.mockito.internal.listeners.StubbingLookupListener;
+import org.mockito.invocation.Invocation;
+import org.mockito.invocation.MatchableInvocation;
+import org.mockito.quality.Strictness;
+import org.mockito.stubbing.Stubbing;
+
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.List;
+
+import static org.mockito.Mockito.mockingDetails;
+
+/**
+ * Default implementation of stubbing lookup listener.
+ * Fails early if stub called with unexpected arguments, but only if current strictness is set to STRICT_STUBS.
+ */
+class DefaultStubbingLookupListener implements StubbingLookupListener {
+
+    Strictness currentStrictness;
+
+    DefaultStubbingLookupListener(Strictness strictness) {
+        this.currentStrictness = strictness;
+    }
+
+    public void onStubbingLookup(Invocation invocation, MatchableInvocation stubbingFound) {
+        if (currentStrictness != Strictness.STRICT_STUBS) {
```

Comment:
Instead of passing the Strictness to this class, I think it is better to remove it from here and when this class is used add the conditional. E.g. `if (currentStrictness == Strictness.STRICT_STUBS)` then add this listener to the CreationSettings